dev/reference/security/role-based-access-control.md

```diff
@@ -0,0 +1,412 @@
++-------------------------------------------------------------+
+```
+
+{{< copyable "sql" >}}
+
+```sql
+SHOW GRANTS FOR 'u1'@'localhost' USING 'r1', 'r2';
+```
+
+```
++---------------------------------------------------------------------+
+| Grants for u1@localhost                                             |
++---------------------------------------------------------------------+
+| GRANT USAGE ON *.* TO `u1`@`localhost`                              |
+| GRANT Select, Insert, Update, Delete ON `db1`.* TO `u1`@`localhost` |
+| GRANT `r1`@`%`,`r2`@`%` TO `u1`@`localhost`                         |
++---------------------------------------------------------------------+
+```
+
+You can use `SHOW GRANTS` or `SHOW GRANTS FOR CURRENT_USER()` to check the current user's privileges. There is a difference between these two statements:
+
+- `SHOW GRANTS` shows the privilege of the enabled role for the current user.
+- `SHOW GRANTS FOR CURRENT_USER()` does not show the enabled role's privilege.
+
+### Authorization table
+
+In addition to four system [privilege tables](/reference/security/privilege-system.md#privilege-table), the RBAC system introduces two new system privilege tables:
+
+- `mysql.role_edges`: records the authorization relationship of the role and user.
+- `mysql.default_roles`: records default roles of each user.
+
+#### `mysql.role_edges`
+
+`mysql.role_edges` contains the following data:
+
+{{< copyable "sql" >}}
+
+```sql
+select * from mysql.role_edges;
+```
+
+```
++-----------+-----------+---------+---------+-------------------+
+| FROM_HOST | FROM_USER | TO_HOST | TO_USER | WITH_ADMIN_OPTION |
++-----------+-----------+---------+---------+-------------------+
+| %         | r_1       | %       | u_1     | N                 |
++-----------+-----------+---------+---------+-------------------+
+1 row in set (0.00 sec)
+```
+
+- `FROM_HOST` and `FROM_USER` indicate the role's host name and user name respectively.
+- `TO_HOST` and `TO_USER` indicate the host name and user name of the user to which a role is granted.
+
+#### `mysql.default_roles`
+
+`mysql.default_roles` shows which roles have been enabled by default for each user.
+
+{{< copyable "sql" >}}
+
+```sql
+select * from mysql.default_roles;
+```
+
+```
++------+------+-------------------+-------------------+
+| HOST | USER | DEFAULT_ROLE_HOST | DEFAULT_ROLE_USER |
++------+------+-------------------+-------------------+
+| %    | u_1  | %                 | r_1               |
+| %    | u_1  | %                 | r_2               |
++------+------+-------------------+-------------------+
+2 rows in set (0.00 sec)
+```
+
+- `HOST` and `USER` indicate the user's host name and user name respectively.
+- `DEFAULT_ROLE_HOST` and `DEFAULT_ROLE_USER` indicate the host name and user name of the default role respectively.
+
+### References
+
+Because RBAS, user management, and privilege management are closely related, you can refer to operation details in the following resources:
```

Comment:
```suggestion
Because RBAC, user management, and privilege management are closely related, you can refer to operation details in the following resources:
```